### PR TITLE
OP-1748: Support DNS resolution in new networking component

### PR DESCRIPTION
**Pull request by marc-casperlabs**
Tested this in a small local testnet, this lets me use DNS names in bind_address, public_address and known_addresses without issues here.

### CODE REVIEW

**Review by rob-casper — APPROVED** [review]

File: node/src/components/network.rs [hunk 1]
```diff
@@ -700,13 +700,30 @@ fn handle_identify_event(swarm: &mut Swarm<Behavior>, event: IdentifyEvent) {
...
-        "/ip4/{}/tcp/{}",
-        parts_itr.next().expect("address should contain IP segment"),
-        parts_itr
-            .next()
-            .expect("address should contain port segment")
-    );
+    let multiaddr_str = if address
+        .chars()
+        .next()
+        .expect("cannot convert emptry address")
```
Comment (Fraser999) [inline node/src/components/network.rs:706]:
```suggestion
        .expect("cannot convert empty address")
```

File: node/src/components/network.rs [hunk 2]
```diff
@@ -700,13 +700,30 @@ fn handle_identify_event(swarm: &mut Swarm<Behavior>, event: IdentifyEvent) {
...
+        .is_numeric()
```
Comment (Fraser999) [inline node/src/components/network.rs:707]: This isn't robust: hostnames could begin with a digit.  I'm expecting this whole function to become obsolete once we remove the small_network component though and start using mutiaddr format in our config.  In which case, I'm ok with this for now.

**Review by Fraser999 — APPROVED** [review]
Optimistic approval given that we agree that `address_str_to_multiaddr` should be removed soon.

File: node/src/components/network.rs (hunk 2 above)
Comment (marc-casperlabs) [reply node/src/components/network.rs:707]: I've had the same thought a few hours after submitting this. I'm also in the "this should be gone soon" camp. Thanks!

**Pull request merged**